File: CPP_Joc_Windows_Android/SH_Client_Win_Cpp_Cmake/App/src/base/menu/util/MenuItemConfig_Data.cpp
#include "MenuItemConfig_Data.h"

MenuItemConfig_Data::MenuItemConfig_Data(std::string itemID, VisualPropertiesConfig* itemProperties)
    : super(itemID),
      itemProperties(itemProperties)
{
    //void
}

VisualPropertiesConfig* MenuItemConfig_Data::getItemProperties() {
    return itemProperties;
}

MenuItemConfig_Data::~MenuItemConfig_Data() {
    //void
}

File: Java_Joc_Windows_Android/AndroidStudioProject/core/src/base/menu/util/MenuItemConfig_Data.java
package base.menu.util;

import base.visual.util.VisualPropertiesConfig;

public class MenuItemConfig_Data extends MenuItemConfig implements IMenuItemConfig_DynamicProperties {

	private VisualPropertiesConfig itemProperties;

	public MenuItemConfig_Data(String itemID, VisualPropertiesConfig itemProperties) {
		super(itemID);

		this.itemProperties = itemProperties;
	}

	@Override
	public VisualPropertiesConfig getItemProperties() {
		return itemProperties;
	}
}
